### PR TITLE
adding srt/1.4.2

### PR DESCRIPTION
**Pull request by prince-chrismc**
Specify library name and version:  **srt/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by prince-chrismc** [comment]
> Failure in build 2 (`4942dad6713937bfa0f402678291d31a9a74e387`):
> 
> * **srt/1.4.1**
>   
>   * Hooks errors detected:
>     
>     * `[HOOK - conan-center.py] pre_export(): ERROR: [CONFIG.YML HAS NEW VERSION (KB-H052)] The version "1.4.2" exists in "conandata.yml" but not in "../config.yml", so it will not be built. Please update "../config.yml" to include newly added version "1.4.2". (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H052) `

Best. Hook. Ever.

File: recipes/srt/all/conanfile.py [hunk 1]
```diff
@@ -52,6 +56,8 @@ def _configure_cmake(self):
         self._cmake.definitions["ENABLE_LOGGING"] = False
         self._cmake.definitions["ENABLE_SHARED"] = self.options.shared
         self._cmake.definitions["ENABLE_STATIC"] = not self.options.shared
+        if self._has_stdcxx_sync:
+            self._cmake.definitions["ENABLE_STDCXX_SYNC"] = True
```
Comment (uilianries) [inline recipes/srt/all/conanfile.py:60]:
```suggestion        
        self._cmake.definitions["ENABLE_STDCXX_SYNC"] = self._has_stdcxx_sync
```

You can pass directly. If there no usage for older versions, CMake will only prompt a warning with other definitions.

**Review by uilianries — APPROVED** [review]
LGTM

**Review by SSE4 — APPROVED** [review]

File: recipes/srt/all/conanfile.py [hunk 2]
```diff
@@ -35,8 +39,8 @@ def configure(self):
             del self.options.fPIC
 
     def requirements(self):
-        self.requires("openssl/1.1.1g")
-        if self.settings.os == "Windows":
+        self.requires("openssl/1.1.1h")
+        if self.settings.os == "Windows" and not self._has_stdcxx_sync:
             self.requires("pthreads4w/3.0.0")
```
Comment (SpaceIm) [inline recipes/srt/all/conanfile.py:44]: I guess that MinGW with pthread doesn't need `pthreads4w`.

Comment (prince-chrismc) [reply recipes/srt/all/conanfile.py:44]: I am sorry, ive never used nor do we need mingw 
for production so I never think I about it. 🙊 

I would like this available for my team so I would prefer to tackle  that in the near future so I can open an issue for adding support... would be a good excuse for me to learn

**Review by intelligide — APPROVED** [review]

**Pull request merged**